### PR TITLE
Update resource function generation for design v2 

### PR DESCRIPTION
**Pull request by sahanHe**
## Purpose

Fixes: [3945](https://github.com/ballerina-platform/ballerina-standard-library/issues/3945), [3947](https://github.com/ballerina-platform/ballerina-standard-library/issues/3947), [3948](https://github.com/ballerina-platform/ballerina-standard-library/issues/3948), [3946](https://github.com/ballerina-platform/ballerina-standard-library/issues/3946)

## Checklist
- [x] Linked to an issue
- [ ] ~Updated the specification~
- [ ] ~Updated the changelog~
- [ ] ~Added tests~

### CODE REVIEW

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_11/generated/entities/generated_client.bal [hunk 1]
```diff
@@ -4,24 +4,93 @@
...
     }
     isolated resource function put medicalneed/[int needId](MedicalNeedUpdate value) returns MedicalNeed|persist:Error {
-        return error persist:Error("unsupported operation");
+        _ = check self.persistClients.get("medicalneed").runUpdateQuery({"needId": needId, }, data);
+        return self->/medicalneed/[needId].get();
     }
     isolated resource function delete medicalneed/[int needId]() returns MedicalNeed|persist:Error {
-        return error persist:Error("unsupported operation");
+        MedicalNeed 'object = check self->/medicalneed/[needId].get();
+        _ = check self.persistClients.get("medicalneed").runDeleteQuery({"needId": needId, });
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_11/generated/entities/generated_client.bal:52]:
```suggestion
        _ = check self.persistClients.get("medicalneed").runDeleteQuery({"needId": needId });
```

Comment (daneshk) [reply persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_11/generated/entities/generated_client.bal:52]: check all places

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_11/generated/entities/generated_client.bal [hunk 2]
```diff
@@ -4,24 +4,93 @@
...
+import ballerinax/mysql;
 
 public client class EntitiesClient {
-    *persist:AbstractPersistClient;
+
+    private final mysql:Client dbClient;
+
+    private final map<persist:SQLClient> persistClients;
+
+    private final map<persist:Metadata> metadata = {medicalneed: {entityName: "MedicalNeed", tableName: 'MedicalNeed ', needId: {columnName: "needId", 'type: int}, itemId: {columnName: "itemId", 'type: int}, beneficiaryId: {columnName: "beneficiaryId", 'type: int}, period: {columnName: "period", 'type: time:Civil}, urgency: {columnName: "urgency", 'type: string}, quantity: {columnName: "quantity", 'type: int} keyFields: ["needId"]}};
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_11/generated/entities/generated_client.bal:16]: Shall we format this code more readable manner

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_11/generated/entities/generated_client.bal [hunk 3]
```diff
@@ -4,24 +4,93 @@
...
+
+    public function init() returns persist:Error? {
+        self.dbClient = check new (host = host, user = user, password = password, database = database, port = port);
+        self.persistClients = {medicalneed: check new (self.dbClient, self.metadata.get("medicalneed").entityName, self.metadata.get("medicalneed").tableName, self.metadata.get("medicalneed").keyFields, self.metadata.get("medicalneed").fieldMetadata)};
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_11/generated/entities/generated_client.bal:20]: Shall we format this code more readable manner

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_11/generated/entities/generated_client.bal [hunk 4]
```diff
@@ -4,24 +4,93 @@
...
+            return new stream<MedicalNeed, persist:Error?>(new MedicalNeedStream(result));
+        }
     }
     isolated resource function get medicalneed/[int needId]() returns MedicalNeed|persist:Error {
-        return error persist:Error("unsupported operation");
+        return (check self.persistClients.get("medicalneed").runReadByKeyQuery(MedicalNeed, needId)).cloneWithType(MedicalNeed);
     }
-    isolated resource function post medicalneed(MedicalNeedInsert[] data) returns int[]|persist:Error {
-        return error persist:Error("unsupported operation");
+    isolated resource function post medicalneed(MedicalNeedInsert[] data) returns [int][]|persist:Error {
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_11/generated/entities/generated_client.bal:41]:
```suggestion
    isolated resource function post medicalneed(MedicalNeedInsert[] data) returns int[]|persist:Error {
```

Comment (daneshk) [reply persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_11/generated/entities/generated_client.bal:41]: we only need to consider tuple, when there is a composite primary key

File: persist-cli-tests/src/test/java/io/ballerina/persist/tools/utils/GeneratedSourcesTestUtils.java [hunk 5]
```diff
@@ -77,16 +74,16 @@ public static void assertGeneratedSources(String subDir, Command cmd) {
...
-            Assert.assertFalse(compilation.diagnosticResult().hasErrors());
-        }
+//        if (!(subDir.equals("tool_test_generate_4") || subDir.equals("tool_test_generate_26") ||
+//                subDir.equals("tool_test_generate_18"))) {
+
+//            BuildProject buildProject = BuildProject.load(Paths.get(GENERATED_SOURCES_DIRECTORY).resolve(subDir)
+//                    .toAbsolutePath());
+//            Package currentPackage = buildProject.currentPackage();
+//            PackageCompilation compilation = currentPackage.getCompilation();
+//            Assert.assertFalse(compilation.diagnosticResult().hasErrors());
```
Comment (daneshk) [inline persist-cli-tests/src/test/java/io/ballerina/persist/tools/utils/GeneratedSourcesTestUtils.java:84]: Any reason to remove the build after client generation?

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_13/generated/entities/generated_client.bal [hunk 6]
```diff
@@ -4,56 +4,226 @@
 // It should not be modified by hand.
 
 import ballerina/persist;
+import ballerina/sql;
+import ballerinax/mysql;
+import ballerina/time;
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_13/generated/entities/generated_client.bal:9]: order the import in alphabetical order

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_11/generated/entities/generated_client.bal (hunk 3 above)
Comment (sahanHe) [reply persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_11/generated/entities/generated_client.bal]: updated in [aa95e9c](https://github.com/ballerina-platform/persist-tools/pull/106/commits/aa95e9c0a62f306236f4dc03ae1d6f60593a1b6e)

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_11/generated/entities/generated_client.bal (hunk 2 above)
Comment (sahanHe) [reply persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_11/generated/entities/generated_client.bal]: updated in [aa95e9c](https://github.com/ballerina-platform/persist-tools/pull/106/commits/aa95e9c0a62f306236f4dc03ae1d6f60593a1b6e)

File: persist-cli-tests/src/test/java/io/ballerina/persist/tools/utils/GeneratedSourcesTestUtils.java (hunk 5 above)
Comment (sahanHe) [reply persist-cli-tests/src/test/java/io/ballerina/persist/tools/utils/GeneratedSourcesTestUtils.java:84]: We couldn't enable this until all the required changes are done from the moudle-persist

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_13/generated/entities/generated_client.bal (hunk 6 above)
Comment (sahanHe) [reply persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_13/generated/entities/generated_client.bal]: updated in [aa95e9c](https://github.com/ballerina-platform/persist-tools/pull/106/commits/aa95e9c0a62f306236f4dc03ae1d6f60593a1b6e)

File: persist-cli-tests/src/test/java/io/ballerina/persist/tools/utils/GeneratedSourcesTestUtils.java (hunk 5 above)
Comment (kalaiyarasiganeshalingam) [reply persist-cli-tests/src/test/java/io/ballerina/persist/tools/utils/GeneratedSourcesTestUtils.java]: Enabled now in [e31880d](https://github.com/ballerina-platform/persist-tools/pull/106/commits/e31880d0f4165a2a2c2e1390f97188025e03fbe9)

**Review by daneshk — APPROVED** [review]

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_24/generated/rainier/generated_client.bal [hunk 7]
```diff
@@ -1,34 +1,382 @@
...
+
+    isolated resource function put department/[string deptNo](DepartmentUpdate value) returns Department|persist:Error {
+        _ = check self.persistClients.get(DEPARTMENT).runUpdateQuery({"deptNo": deptNo}, value);
+        return self->/department/[deptNo].get();
+    }
+
+    isolated resource function delete department/[string deptNo]() returns Department|persist:Error {
+        Department 'object = check self->/department/[deptNo].get();
+        _ = check self.persistClients.get(DEPARTMENT).runDeleteQuery({"deptNo": deptNo});
+        return 'object;
```
Comment (niveathika) [inline persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_24/generated/rainier/generated_client.bal:151]:
```suggestion
        Department result= check self->/department/[deptNo].get();
        _ = check self.persistClients.get(DEPARTMENT).runDeleteQuery({"deptNo": deptNo});
        return result;
```

Lets coordinate with @kaneeldias and change this name if possible

Comment (daneshk) [reply persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_24/generated/rainier/generated_client.bal:151]: @kalaiyarasiganeshalingam shall we fix this

Comment (kalaiyarasiganeshalingam) [reply persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_24/generated/rainier/generated_client.bal:151]: Updated in [7e5dedd](https://github.com/ballerina-platform/persist-tools/pull/106/commits/7e5deddedf6e0aa89f1951105e0f2c921725628c)

**Review by daneshk — APPROVED** [review]

**Comment by daneshk** [comment]
@niveathika please review and merge

**Review by niveathika — APPROVED** [review]

**Pull request merged**